src/api/app/controllers/webui/kiwi/images_controller.rb

```diff
@@ -105,6 +106,23 @@ def cleanup_non_project_repositories!
         @image.repositories.delete_all
         params[:kiwi_image].delete(:repositories_attributes)
       end
+
+      def image_error_messages(title, image, packages)
+        message = { title: title }
+        message["Image Errors:"] = image.errors.messages[:base] if image.errors.messages[:base].present?
+
+        { repository: image.repositories, package: packages }.each do |key, records|
+          records.each do |record|
+            if record.errors.present?
+              message["#{key.capitalize}: #{record.name}"] ||= []
+              message["#{key.capitalize}: #{record.name}"] << record.errors.messages.map { |_key, value| value }
+              message["#{key.capitalize}: #{record.name}"].flatten!
```

Comment:
Why is this method necessary? Can we not just use the errors object/hash that ActiveRecord gives us?

Reply:
Because the error hash gives us 'Repository[0]...` and I'm trying to do this more understandable grouping all the errors related to each element.